diff --git a/components/bt/bluedroid/stack/l2cap/include/l2c_int.h b/components/bt/bluedroid/stack/l2cap/include/l2c_int.h
index e2c0ef6a7..62205d253 100644
--- a/components/bt/bluedroid/stack/l2cap/include/l2c_int.h
+++ b/components/bt/bluedroid/stack/l2cap/include/l2c_int.h
@@ -48,26 +48,26 @@
 /* Timeouts. Since L2CAP works off a 1-second list, all are in seconds.
 */
 #define L2CAP_LINK_ROLE_SWITCH_TOUT  10           /* 10 seconds */
-#define L2CAP_LINK_CONNECT_TOUT      60           /* 30 seconds */
-#define L2CAP_LINK_CONNECT_TOUT_EXT  120          /* 120 seconds */
-#define L2CAP_ECHO_RSP_TOUT          30           /* 30 seconds */
+#define L2CAP_LINK_CONNECT_TOUT      2            /* 30 seconds */
+#define L2CAP_LINK_CONNECT_TOUT_EXT  10           /* 120 seconds */
+#define L2CAP_ECHO_RSP_TOUT          10           /* 30 seconds */
 #define L2CAP_LINK_FLOW_CONTROL_TOUT 2            /* 2  seconds */
-#define L2CAP_LINK_DISCONNECT_TOUT   30           /* 30 seconds */
+#define L2CAP_LINK_DISCONNECT_TOUT   10           /* 30 seconds */
 
 #ifndef L2CAP_CHNL_CONNECT_TOUT      /* BTIF needs to override for internal project needs */
 #define L2CAP_CHNL_CONNECT_TOUT      60           /* 60 seconds */
 #endif
 
-#define L2CAP_CHNL_CONNECT_TOUT_EXT  120          /* 120 seconds */
-#define L2CAP_CHNL_CFG_TIMEOUT       30           /* 30 seconds */
+#define L2CAP_CHNL_CONNECT_TOUT_EXT  10           /* 120 seconds */
+#define L2CAP_CHNL_CFG_TIMEOUT       10           /* 30 seconds */
 #define L2CAP_CHNL_DISCONNECT_TOUT   10           /* 10 seconds */
 #define L2CAP_DELAY_CHECK_SM4        2            /* 2 seconds */
 #define L2CAP_WAIT_INFO_RSP_TOUT     3            /* 3 seconds */
 #define L2CAP_WAIT_UNPARK_TOUT       2            /* 2 seconds */
 #define L2CAP_LINK_INFO_RESP_TOUT    2            /* 2 seconds */
 #define L2CAP_UPDATE_CONN_PARAM_TOUT 6            /* 6 seconds */
-#define L2CAP_BLE_LINK_CONNECT_TOUT  30           /* 30 seconds */
-#define L2CAP_BLE_CONN_PARAM_UPD_TOUT   30        /* 30 seconds */
+#define L2CAP_BLE_LINK_CONNECT_TOUT  2           /* 30 seconds */
+#define L2CAP_BLE_CONN_PARAM_UPD_TOUT   10        /* 30 seconds */
 
 /* quick timer uses millisecond unit */
 #define L2CAP_DEFAULT_RETRANS_TOUT   2000         /* 2000 milliseconds */
