diff --git a/components/bt/bluedroid/stack/include/stack/btm_ble_api.h b/components/bt/bluedroid/stack/include/stack/btm_ble_api.h
index 41f9d8bac..5203de9ec 100644
--- a/components/bt/bluedroid/stack/include/stack/btm_ble_api.h
+++ b/components/bt/bluedroid/stack/include/stack/btm_ble_api.h
@@ -178,7 +178,7 @@ typedef UINT8   tBTM_BLE_SFP;
 
 /* default supervision timeout */
 #ifndef BTM_BLE_CONN_TIMEOUT_DEF
-#define BTM_BLE_CONN_TIMEOUT_DEF    600
+#define BTM_BLE_CONN_TIMEOUT_DEF    100
 #endif
 
 /* minimum acceptable connection interval */
diff --git a/components/bt/bluedroid/stack/l2cap/include/l2c_int.h b/components/bt/bluedroid/stack/l2cap/include/l2c_int.h
index e2c0ef6a7..a7c2d5d80 100644
--- a/components/bt/bluedroid/stack/l2cap/include/l2c_int.h
+++ b/components/bt/bluedroid/stack/l2cap/include/l2c_int.h
@@ -66,7 +66,7 @@
 #define L2CAP_WAIT_UNPARK_TOUT       2            /* 2 seconds */
 #define L2CAP_LINK_INFO_RESP_TOUT    2            /* 2 seconds */
 #define L2CAP_UPDATE_CONN_PARAM_TOUT 6            /* 6 seconds */
-#define L2CAP_BLE_LINK_CONNECT_TOUT  30           /* 30 seconds */
+#define L2CAP_BLE_LINK_CONNECT_TOUT  1            /* 1 seconds */
 #define L2CAP_BLE_CONN_PARAM_UPD_TOUT   30        /* 30 seconds */
 
 /* quick timer uses millisecond unit */
diff --git a/components/bt/bluedroid/stack/l2cap/l2c_api.c b/components/bt/bluedroid/stack/l2cap/l2c_api.c
index 5637cac64..92d74874a 100644
--- a/components/bt/bluedroid/stack/l2cap/l2c_api.c
+++ b/components/bt/bluedroid/stack/l2cap/l2c_api.c
@@ -1842,10 +1842,10 @@ UINT16 L2CA_SendFixedChnlData (UINT16 fixed_cid, BD_ADDR rem_bda, BT_HDR *p_buf)
 
 BOOLEAN L2CA_CheckIsCongest(UINT16 fixed_cid, UINT16 handle)
 {
-    tL2C_LCB *p_lcb;
-    p_lcb = l2cu_find_lcb_by_handle(handle);
+    tL2C_LCB *p_lcb = l2cu_find_lcb_by_handle(handle);
+    size_t chnl = fixed_cid - L2CAP_FIRST_FIXED_CHNL;
 
-    if (p_lcb != NULL) {
+    if (p_lcb != NULL && p_lcb->p_fixed_ccbs[chnl] != NULL) {
         return p_lcb->p_fixed_ccbs[fixed_cid - L2CAP_FIRST_FIXED_CHNL]->cong_sent;
     }
 
diff --git a/tools/ldgen/ldgen.py b/tools/ldgen/ldgen.py
index bb70950d1..23b02629e 100755
--- a/tools/ldgen/ldgen.py
+++ b/tools/ldgen/ldgen.py
@@ -92,10 +92,10 @@ def main():
 
         script_model.write(output_file)
 
-    except Exception, e:
+    except Exception as e:
         print("linker script generation failed for %s\nERROR: %s" % (input_file.name, e.message))
         # Delete the file so the entire build will fail; and not use an outdated script.
         os.remove(output_file.name)
 
 if __name__ == "__main__":
-    main()
\ No newline at end of file
+    main()
